tests/test_browser.py

```diff
@@ -3556,6 +3556,41 @@ def test_dynamic_link_many(self):
     self.btest_exit(self.in_dir('main.c'), '3',
                     args=['-s', 'MAIN_MODULE', '--pre-js', 'pre.js'])
 
+  def test_dynamic_link_pthread_many(self):
+    # test asynchronously loading two side modules during startup
+    # they should always load in the same order
+    create_test_file('main.cpp', r'''
+      #include <assert.h>
+      #include <thread>
+      extern "C" int side1();
+      extern "C" int side2();
+      int main() {
+        auto side1_ptr = &side1;
+        auto side2_ptr = &side2;
+        std::thread([=]{
+          REPORT_RESULT(int(
+            side1_ptr() == 1 &&
+            side2_ptr() == 2
+          ));
+        }).detach();
+        return 0;
```

Comment:
Can you `join()` the thread and then `return result`?   (and then as alon mentioned you should be able to use `btest_exit`...  That will require `PROXY_TO_PTHREAD` too.. but that should be fine I think.

Reply:
Running with `PROXY_TO_PTHREAD` all the function pointer are obtained and used in workers (where the libraries are loaded synchronously) and the problem isn't hit. I need to involve the main thread to hit the issue, where I can't `.join()`.
I guess I could obtain the function pointers using some variant of `emscripten_sync_run_in_main_thread`. Would that be preferred?

Reply:
No, that make sense.   Perhaps add a comment as to why this has to run on the main thread and can't use join?